### PR TITLE
Pd 11159 refactor nodetool usage

### PR DESCRIPTION
**Pull request by jpatrickr7**
Made changes to the methods, and way it gets the gossip and thrift status.   Feel free to pick apart and let me know if there can be any changes made.

### CODE REVIEW

File: lib/cassandra/utils/stats/health.rb [hunk 1]
```diff
@@ -22,6 +22,34 @@ def metric_name
...
+            @nodetool_info ||= DaemonRunner::ShellOut.new(command: 'nodetool info')
+            @nodetool_info.run!
+            @nodetool_info.stdout
+          end
+
+
+        def gossipstate
+          results = (nodetool_info || '').split("\n")
+          results.map! { |line| line.strip }
+          results.select! { |line| line.include? 'Gossip active    :' }
```
[the remark this answers is not in the document]
Comment (jpatrickr7) [reply lib/cassandra/utils/stats/health.rb]: So for this just sort of go for it finding the glob of Gossip active? like via Regex or a match?

File: lib/cassandra/utils/stats/health.rb [hunk 2]
```diff
@@ -22,6 +22,34 @@ def metric_name
...
+          results.map! { |line| line.split(':')[1] }
+          results.compact!
+          return nil if results.size != 1
+          results.first.strip.downcase.to_sym
```
[the remark this answers is not in the document]
Comment (jpatrickr7) [reply lib/cassandra/utils/stats/health.rb]: and just drop the .to_sym off the end, or other things required?

File: lib/cassandra/utils/stats/health.rb [hunk 3]
```diff
@@ -22,6 +22,34 @@ def metric_name
...
+        end
+
+        def thriftstate
+          results = (nodetool_info || '').split("\n")
+          results.map! { |line| line.strip }
+          results.select! { |line| line.include? 'Thrift active    :' }
```
[the remark this answers is not in the document]
Comment (jpatrickr7) [reply lib/cassandra/utils/stats/health.rb]: Same as above with gossip (same Q)

File: test/cassandra/health_check_test.rb [hunk 4]
```diff
@@ -8,8 +8,8 @@
   describe :run! do
     it 'succeeds if node is NORMAL and thrift and gossip are running' do
       @checker.stub :nodetool_netstats, 'Mode: NORMAL' do
-        @checker.stub :nodetool_statusthrift, 'running' do
-          @checker.stub :nodetool_statusgossip, 'running' do
+        @checker.stub :thriftstate, 'true' do
+          @checker.stub :gossipstate, 'true' do
```
[the remark this answers is not in the document]
Comment (jpatrickr7) [reply test/cassandra/health_check_test.rb]: okay I think I grok this.

File: lib/cassandra/utils/stats/health.rb [hunk 5]
```diff
@@ -16,6 +16,27 @@ def metric_name
           'cassandra.service.running'
         end
 
+
+        def gossipstate
```
Comment (athompson-r7) [inline lib/cassandra/utils/stats/health.rb]: Can we add some yardoc?

File: lib/cassandra/utils/stats/health.rb [hunk 6]
```diff
@@ -16,6 +16,27 @@ def metric_name
...
+          results = (nodetool_info || '').split("\n")
+          results.map! { |line| line.strip }
+          results.select! { |line| line.include? 'Gossip active' }
+          results.map! { |line| line.split(':')[1] }
+          results.compact!
+          return nil if results.size != 1
+          results.first.strip.downcase
+        end
+
+        def thriftstate
```
Comment (athompson-r7) [inline lib/cassandra/utils/stats/health.rb]: Can we add some yardoc?

File: lib/cassandra/utils/stats/health.rb [hunk 7]
```diff
@@ -16,6 +16,27 @@ def metric_name
           'cassandra.service.running'
         end
 
+
+        def gossipstate
+          results = (nodetool_info || '').split("\n")
+          results.map! { |line| line.strip }
```
Comment (athompson-r7) [inline lib/cassandra/utils/stats/health.rb]: Wondering if `nodetool_info` should be responsibe for the `split` and `strip`?

File: lib/cassandra/utils/stats/health.rb [hunk 8]
```diff
@@ -38,25 +59,11 @@ def task_id
...
-        # Run the "nodetool statusthrift' command and return the output
-        #
-        # @return [String, nil] Output from the "nodetool statusthrift" command
-        #
-        def nodetool_statusthrift
-          @nodetool_statusthrift||= DaemonRunner::ShellOut.new(command: 'nodetool statusthrift')
-          @nodetool_statusthrift.run!
-          @nodetool_statusthrift.stdout
-        end
+        def nodetool_info
```
Comment (athompson-r7) [inline lib/cassandra/utils/stats/health.rb]: Can we add some yardoc?

File: lib/cassandra/utils/stats/health.rb (hunk 7 above)
Comment (jpatrickr7) [reply lib/cassandra/utils/stats/health.rb]: it seems logical that it's only having to do it once.  That was the logic that frank extended over to me.

File: lib/cassandra/utils/stats/health.rb (hunk 6 above)
Comment (jpatrickr7) [reply lib/cassandra/utils/stats/health.rb]: added, but need critque

File: lib/cassandra/utils/stats/health.rb (hunk 8 above)
Comment (jpatrickr7) [reply lib/cassandra/utils/stats/health.rb]: added, comments needed.

File: lib/cassandra/utils/stats/health.rb (hunk 5 above)
Comment (jpatrickr7) [reply lib/cassandra/utils/stats/health.rb]: Added some, feel free to crituqe.

**Comment by jpatrickr7** [comment]
Added Yardoc, and some reasoning.  Feel free to critique.

**Comment by jpatrickr7** [comment]
Made more changes.. Yardoc bits correct this time.

**Pull request merged**